x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -543,38 +548,57 @@ export class RulesClient {
 
     // default duration of instance summary is 60 * rule interval
     const dateNow = new Date();
-    const durationMillis = parseDuration(rule.schedule.interval) * 60;
+    const durationMillis = parseDuration(rule.schedule.interval) * (numberOfExecutions ?? 60);
     const defaultDateStart = new Date(dateNow.valueOf() - durationMillis);
     const parsedDateStart = parseDate(dateStart, 'dateStart', defaultDateStart);
 
     const eventLogClient = await this.getEventLogClient();
 
     this.logger.debug(`getAlertSummary(): search the event log for rule ${id}`);
     let events: IEvent[];
+    let executionEvents: IEvent[];
+
     try {
-      const queryResults = await eventLogClient.findEventsBySavedObjectIds(
-        'alert',
-        [id],
-        {
-          page: 1,
-          per_page: 10000,
-          start: parsedDateStart.toISOString(),
-          end: dateNow.toISOString(),
-          sort_order: 'desc',
-        },
-        rule.legacyId !== null ? [rule.legacyId] : undefined
-      );
+      const [queryResults, executionResults] = await Promise.all([
+        eventLogClient.findEventsBySavedObjectIds(
+          'alert',
+          [id],
+          {
+            page: 1,
+            per_page: 10000,
+            start: parsedDateStart.toISOString(),
+            sort_order: 'desc',
+            end: dateNow.toISOString(),
+          },
+          rule.legacyId !== null ? [rule.legacyId] : undefined
+        ),
+        eventLogClient.findEventsBySavedObjectIds(
+          'alert',
+          [id],
+          {
+            page: 1,
+            per_page: numberOfExecutions ?? 60,
+            filter: 'event.provider: alerting AND event.action:execute',
+            sort_order: 'desc',
```

Comment:
We fetch this in `desc` order, but then [we reverse the list](https://github.com/elastic/kibana/pull/122595/files#diff-91fad783d257caf139644bbf4f6784cf730ad728307b37d9350baad691d29522R90) later? Why not fetch in `asc` order?

Reply:
We want to get the newest event log events but post-process them in order from oldest to newest.

Reply:
Ahh, that makes sense, thanks!